src/components/MultipleSelect/index.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useState } from 'react';
+import './index.scss';
+interface MultipleSelectProps<T> {
+   options: {
+      text: string;
+      value: T;
+   }[];
+   value: T[];
+   onChange: (item: T[]) => void;
+   title: string;
+}
+function MultipleSelect<T>({
+   options,
+   value,
+   onChange,
+   title,
+}: MultipleSelectProps<T>) {
+   const [show, setShow] = useState<boolean>(false);
+   const selectedOptions = React.useMemo<
+      {
+         text: string;
+         value: T;
+      }[]
+   >(() => {
+      return options.filter((option) => value.includes(option.value));
+   }, [options, value]);
+
+   const handleSelectItem = (item: { text: string; value: T }) => {
+      const selected: T[] = [];
+      if (selectedOptions.includes(item)) {
+         selectedOptions
+            .filter((value) => value !== item)
+            .map((option) => selected.push(option.value));
+
+         onChange(selected);
+      } else {
+         selectedOptions.push(item);
+         selectedOptions.map((option) => selected.push(option.value));
+         onChange(selected);
```

Comment:
- không cần tạo biến mới nha e, có thể onChange trực tiếp được
- `Array.map` là để trả về modify array, e dùng loop để push thì nên dùng `Array.forEach`
```suggestion
         onChange(value.filter((_value) => _value !== item.value));
      } else {
         onChange([...value, item.value ]);
```